Docs: v24 core (#1068)
pomerium/documentation: content/docs/deploy/core/upgrading.mdx

### Breaking

#### Set Authorization Header

The deprecated `set_authorization_header` configuration setting has been removed. You can use the [Set Request Headers](/docs/reference/routes/headers#set-request-headers) setting to pass IdP tokens to upstream services in any header.

#### Base64-encoded Certificates

Previously, the `certificates` key supported base64-encoded certificates as a value (this option was not documented). We've removed support for base64-encoded certificates for this setting and now require that you only use the relative file location.

**Note:** The `certificates` key is distinct from the `certificate` key. The `certificate` key setting still supports base64-encoded certificates; the `certificates` _list_ does not.

See [Certificates](/docs/reference/certificates#certificates) for more information.

#### Redis Storage Backend

PostgreSQL has been the recommended databroker storage backend since v0.18. Support for Redis has now been removed.

See [Persistence](/docs/internals/data-storage#postgres) for more information.

### Performance

v0.24.0 includes several performance enhancements for Pomerium Core. See the sections below for more information.

> #### Databroker state sync

Reviewer: I think this is also a console-specific change? @wasaga, can you confirm?

Author: Yeah I wasn't sure about this one. I can remove if it is just Console.

> #### Databroker state sync

Author: ```suggestion
```

> Previously, on every sync tick we would request all configuration chunks one by one in order to detect what need be changed (and what has to be deleted). In v24 we have a single databroker syncer.

Author: ```suggestion
```

# Upgrade guide

## 0.24.0

### Breaking

#### Set Authorization Header

The deprecated `set_authorization_header` configuration setting has been removed. You can use the [Set Request Headers](/docs/reference/routes/headers#set-request-headers) setting to pass IdP tokens to upstream services in any header.

> #### Base64-encoded Certificates

Reviewer: This is a little subtle: if I understand correctly, base64-encoded certificates are still supported under the `certificate` key, just not the `certificates` list. Is there some way we can clarify this heading, to avoid alarming anyone using the supported `certificate` setting?

> Previously, the `certificates` key supported base64-encoded certificates as a value. We've removed support for base64-encoded certificates for this setting and now require that you only use the relative file location.

Reviewer: We might want to emphasize that this was an undocumented feature?

See [Certificates](/docs/reference/certificates#certificates) for more information.

> #### Remove Redis Storage Backend

Reviewer: ```suggestion
#### Redis Storage Backend
```

> Support for Redis as a storage backend for Pomerium has been completely removed. We recommend a dedicated Postgres instance for Pomerium to provide the strongest security and performance guarantees.

Reviewer: Should we mention that the Redis backend has been discouraged since v0.18?

```suggestion
PostgreSQL has been the recommended databroker storage backend since v0.18. Support for Redis has now been removed.
```

See [Persistence](/docs/internals/data-storage#postgres) for more information.

### Performance

v0.24.0 includes several performance enhancements for Pomerium Core. See the sections below for more information.

> #### Smaller configuration chunks

Reviewer: I think we should remove the Console-specific items from this section.
